docs/troubleshooting-guide/sync-fingerprint-scan.md

```diff
@@ -1,85 +1,184 @@
-3. If you've just added a table or a column, Metabase might not have synced yet. You can manually run the sync process by going to the Admin Panel, selecting "Databases", choosing your database, and clicking on "Sync database schema now".
-4. To see if the problem is caused by lack of database privileges, try running a query like the one below for each table you think you should be able to access :
+A sync query should show up like this in your database's query execution table (using whatever role is defined in your database connection string):
 
+```sql
+SELECT TRUE
+FROM "your_database"
+WHERE 1 <> 1
+LIMIT 0
 ```
-SELECT *
-FROM table
-LIMIT 1
+
+If you’ve just set up a new database in Metabase, the initial sync query might still be running—--especially if you have a large database. To run the query, Metabase must:
+
+- successfully connect to your database, and
+- be granted permissions to query that database.
+
+If the connection is failing, or the database privileges are wrong, the sync query won't be able to run. This will block the _initial_ scan and fingerprinting queries as well.
+
+## Syncing new or updated tables and views
+
+1. Go to **Admin** > **Troubleshooting** > **Logs** to check the status of the sync.
+2. If the logs show you that the sync is failing on a specific table or view, make sure Metabase has the correct database privileges to query that table or view.
+3. Run a query against your database from the Metabase SQL editor to check for database connection or database privilege errors that aren't listed in the logs:
+
+    ```sql
+    SELECT *
+    FROM "your_database"."your_schema"."your_table_or_view"
+    LIMIT 1
+    ```
+4. [Manually re-sync](../databases/connecting.md#manually-syncing-tables-and-columns) the table or view.
+
+**Explanation**
+
+Metabase needs the correct database privileges to run the sync query against a given database, schema, and table (or view):
+
+```sql
+SELECT TRUE
+FROM "your_database"."your_schema"."your_table_or_view"
+WHERE 1 <> 1
+LIMIT 0
 ```
 
-Note that we only get the first 10,000 documents when scanning a MongoDB collection, so if you're not seeing some new fields, those fields might not exist in the documents we looked at. Please see [this discussion][metabase-mongo-missing] for more details.
+## Syncing tables with JSON records
+
+1. Go to **Admin** > **Databases** > **your database** > **Show advanced options**.
+2. Click **Disable "JSON unfolding"**
+3. Click **Save changes**.
+4. Click **Sync database schema**.
+
+**Explanation**
+
+Metabase will try to unfold JSON and JSONB records during the sync process, which can take up a decent chunk of query execution time. If you have a lot of JSON records, try disabling the automatic unfolding option to pull the sync out of slow-motion. Remember that you can follow the status of the sync from **Admin** > **Troubleshooting** > **Logs**.
+
+## Scanning
```

Comment:
This is missing the key information that was previously available in https://github.com/metabase/metabase/pull/26572/files#diff-073867a29b01598c6a1a7c2463296598789a7ccdb94c4f0daa139e3ab09066aeL13
> When you set a field to "A list of all values" in the [Data Model](../data-modeling/metadata-editing.md), which is used to display options in dropdown menus, scanning looks at the first 1,000 distinct records (ordered ascending). For each field scanned, Metabase stores only the first 100 kilobytes of text. If more values exist, Metabase displays the stored values in the dropdown menus, and only triggers a database search query to look for more values when people type in the search box for that filter widget

The key being only 1000 values will be shown. It is recommended to use Search if there are more values.

Reply:
Thanks! I copied some of this over to the reference doc a while back as well (must have thought I copied everything).
https://www.metabase.com/docs/latest/databases/connecting#how-database-scans-work

I'll make sure the key point is kept in the troubleshooting explanation here, and we'll store the additional details in the reference doc.

Reply:
Yes, I know, but that's on a different page. This is the troubleshooting page and should include all information.

Reply:
Got it, I'll preserve as much as possible. My original thought was to make things more maintainable from a doc perspective. For example, if we change the name of the option in the Data Model from "A list of all values" to "dropdown menu", then we only need to update that in the reference docs.

In general, I'm trying to follow a pattern like this:
1. Update troubleshooting guides to refer to use cases
2. Link the use cases to the reference docs, where we keep the specifics (e.g. descriptions of the UI).

The feedback is helpful though. You can tell I tend towards cutting things away first, and adding them back later :)